### PR TITLE
Change in DM helm-install to system app upload/apply

### PR DESCRIPTION
**Pull request by sharbhar**

### CODE REVIEW

**Review by yjian118 — CHANGES_REQUESTED** [review]
Considering the current usage of this playbook, I'll not approve this change unless addressed the following concerns:
1 blocked initial deployment(should be done in bootstrap)
2 properly finished the upgrade/update condition
3 covers the usage of the day-2 operation

**Comment by sharbhar** [comment]
abandoning this commit

**Comment by sharbhar** [comment]
reopening

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:906]: Linefeed character is missing before the end of file (EOF).

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:321]: If we are planning for backward compatibility of this playbook we cannot remove this.
Otherwise, we have to make sure all relevant tasks of helmv2 and helmv3 upgrade / installation is going to be handled by system app framework and such tasks be deleted from the playbook (tasks such as deleting webhook service, restarting DM pod etc.).

**Review by sriram-gn — CHANGES_REQUESTED** [review]
Few more concerns:

1. What if helm chart is a user uploaded artifact? I don't see this being handled.
2. I see old path references for helm / overrides such as:
`/usr/local/share/applications/helm/wind-river-cloud-platform-deployment-manager-*.tgz`

The original playbook is designed to fail if it doesn't find these files in path and user uploaded artifacts is false.
 
Depending upon the plan for backward compatibility these things have to be handled.
Please add them to test cases as well.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:323]: Please use ansible.builtin.file to check the existence of a file

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:322]: Can we use an ansible variable instead of hard-coding the file path? So we can let the StarlingX user overwrite it if required

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:340]: Is 6 seconds enough to upload the tarball?

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:321]: agree

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:377]: This block & condition is not necessary as expected to fail before the start of the block if condition not met

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:314]: Need to re-visit all the tasks with "helmv3_installed" if the application no longer expected to be exist in helmv3 cli

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:321]: Need to remove all dm_monitor_playbook related facts/tasks

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:314]: Also request to clean helmv2_installed as no longer supported

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:321]: What's the meaning of this flag? Can you add a comment to describe it?

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:379]: What is the "is_unlock_ready" mean here? Why do we want to check it for this whole block? Can you add some comments above the block or here to describe that?

**Comment by sharbhar** [comment]
Testing in progress.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:230]: It won't reach this task since the previous task won't succeed until `("applied" in app_apply_status.stdout)`.
We might have to ignore_errors in `Wait deployment-manager app to become applied` task with an update to fail task as shown below:

```
       - name: Fail if deployment-manager app apply fails
         fail:
           msg: "The apply of DM system app has failed. Expected 'applied' state but the app is in '{{app_apply_status.stdout}}' state."
         when: ("applied" not in app_apply_status.stdout)

```

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:227]: Please correct the name to: `name: Fail if deployment-manager app apply fails`

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:211]: Please correct the name to `name: Fail if deployment-manager app upload fails`

**Review by sriram-gn — COMMENTED** [review]
Please test some of the failure scenarios as well to make sure it works as expected.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:159]: Please use `--nowrap` with `system` commands otherwise it may be prone to unexpected failures.
`source /etc/platform/openrc; system application-list --nowrap | grep deployment-manager | awk '{ print $10 }'`

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:205]: Please use `--nowrap` with `system` commands otherwise it may be prone to unexpected failures.
`source /etc/platform/openrc; system application-list --nowrap | grep deployment-manager | awk '{ print $10 }'`

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:221]: Please use `--nowrap` with `system` commands otherwise it may be prone to unexpected failures.
`source /etc/platform/openrc; system application-list --nowrap | grep deployment-manager | awk '{ print $10 }'`

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:217]: Fail if anything other than "uploaded".
`when: not ("uploaded" in app_upload_status.stdout)`

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:236]: Fail if anything other than "applied".
when: not ("applied" in app_apply_status.stdout)

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sriram-gn) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:10]: Unresolved this conversation since I did not see any code changes related to this, please respond to the query and then resolve it, thank you.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:160]: Do we need to verify the status? I assume "upload/upload-failed/applying/" should be treated differently

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:238]: May need to update the block for the following cases:

- application was uploaded - can we upload it again?
- application was upload-failed: I assume this block should be also fine in this case
- application was applying - should only waiting for applied
- application was removing:  I assume this block should be also fine in this case, suggest to double check
- application was remove-failed:  I assume this block should be also fine in this case, suggest to double check

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:325]: Expect to preserve this task, please check the pod's name and delete the pod in the case not a fresh install

**Comment by yjian118** [comment]
Please consider all the application status need to handle during day-2 operation and update the procedures accordingly

**Comment by sharbhar** [comment]
Testing completed.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:171]: expect to handle the case the dm_app_status.stdout != 'applied' or deployment_manager_pod_status.stdout != 'Running', we cannot simply set the reconfig flag = False in these cases

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:162]: Suggest using the same commands to wait for the pods to be ready below in concern with the pod in a transitional status

Comment (yjian118) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:162]: Further are there any cases the application is 'applied' but pod is not 'Running'?  I assume this check should be included in the definition of 'applied', but I'm okay with this check.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (yjian118) [inline docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:302]: I'm a bit concerned about why there's a change from the existing commit? Seems from a resolve of merge conflicts. We'll need to monitor the result after the merge

**Comment by yjian118** [comment]
Please update the 1st message to include all the test cases passed with this commit

**Review by yjian118 — APPROVED** [review]
The change looks fine, is there any dependencies we need to care before merging the change?

**Comment by sharbhar** [comment]
> The change looks fine, is there any dependencies we need to care before merging the change?
Checked, didn't find any dependencies to check.
we can merge.

**Comment by sharbhar** [comment]
> Please update the 1st message to include all the test cases passed with this commit

Already present.

File: docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml
Comment (sharbhar) [reply docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml:302]: Will monitor, once the result will be merged.

**Pull request merged**
